nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/validations/ValidationError.java

```diff
@@ -5,7 +5,7 @@
 /**
  * Represents a validation error of a form's input field.
  */
-public abstract class ValidationError {
+public abstract class ValidationError extends Exception {
```

Comment:
I'm not sure if it makes sense to have `ValidationError` extend `Exception`. In my opinion, an Exception makes sense for unexpected results or developer error in which case the developer needs to handle it, since the application might be in an inconsistent state. In this case, bad user input is not something unexpected or a developer error. It is expected that the application might get invalid input.

Was there a technical reason or a benefit to extending `Exception`?

I suggest that the method `InputValidator#validate()` just return an instance of `ValidationError` if there is a validation error, or `null` if there is no error.